nixos/lib/eval-config.nix

```diff
@@ -68,7 +68,19 @@ let
         _module.check = lib.mkDefault check;
       };
     };
-  allUserModules = modules ++ legacyModules;
+
+  allUserModules =
+    let
+      # Add the invoking file as error message location for modules
+      # that don't have their own locations; presumably inline modules.
+      modulesLocation = builtins.unsafeGetAttrPos "modules" evalConfigArgs;
```

Comment:
You could make `modulesLocation` a parameter (in `evalConfigArgs`) so the caller of this file can control this behavior to
 - set it to a better value.
 - disable it if it already calls `setModuleDefaultLocation` by itself. 

A similar effect can already be achieved by only calling `setModuleDefaultLocation` first, as the innermost takes priority, but has the potential to create long-ish chains of imports that are unnecessary and do show up in a few module system error messages.
This is relevant for libraries and tools like NixOps that call NixOS for you and add modules of their own, invalidating the `modules` pos.

Reply:
Done, and tested